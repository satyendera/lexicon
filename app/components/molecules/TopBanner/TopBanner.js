/**
 *
 * TopBanner
 *
 *  */

// @flow
import React, { PureComponent } from 'react';
import styles from './TopBanner.style';
import withStyles from '../../../lib/withStyles';
import type { Props } from './types';

class TopBanner extends PureComponent<Props> {
  typingSpeed = 80;

  deletingSpeed = 30;

  deleteDelay = 2000;

  PART_INDEX = 0;

  PART = 0;

  INTERVAL_VAL = undefined;

  ELEMENT: HTMLElement;

  points: Array<string> = [];

  componentDidMount() {
    const { tickerPoints } = this.props;
    this.points = tickerPoints;
    this.PART = 0;
    this.PART_INDEX = 0;
    if (tickerPoints) {
      this.ELEMENT = document.querySelector(`#ticker`);
      this.INTERVAL_VAL = setInterval(() => {
        this.type();
      }, this.typingSpeed);
    }
  }

  type = () => {
    // Get substring with 1 characater added
    const text = this.points[this.PART].substring(0, this.PART_INDEX + 1);
    this.ELEMENT.innerHTML = text;
    this.PART_INDEX = this.PART_INDEX + 1;

    // If full sentence has been displayed then start to delete the sentence after some time
    if (text === this.points[this.PART]) {
      clearInterval(this.INTERVAL_VAL);
      setTimeout(() => {
        this.INTERVAL_VAL = setInterval(() => {
          this.delete();
        }, this.deletingSpeed);
      }, this.deleteDelay);
    }
  };

  delete = () => {
    // Get substring with 1 characater deleted
    const text = this.points[this.PART].substring(0, this.PART_INDEX - 1);
    this.ELEMENT.innerHTML = text;
    this.PART_INDEX = this.PART_INDEX - 1;
    if (text === '') {
      clearInterval(this.INTERVAL_VAL);
      const pointLength = this.points.length - 1;
      if (this.PART === pointLength) {
        this.PART = 0;
      } else {
        this.PART = this.PART + 1;
      }

      this.PART_INDEX = 0;
      setTimeout(() => {
        this.INTERVAL_VAL = setInterval(() => {
          this.type();
        }, this.typingSpeed);
      }, 500);
    }
  };

  render() {
    const { className, title, subTitle, tickerPoints, subHeading } = this.props;
    return (
      <section className={className} id="banner">
        <div className="inner">
          <h1>{title}</h1>
          {subHeading && <div className="sub-heading">{subHeading}</div>}
          {subTitle && <p>{subTitle}</p>}
          {tickerPoints && <div id="ticker"> </div>}
        </div>
      </section>
    );
  }
}

TopBanner.defaultProps = {};
export default withStyles(TopBanner, styles);
export { TopBanner };
