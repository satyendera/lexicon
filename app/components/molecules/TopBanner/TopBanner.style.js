import { css } from 'styled-components';

export default css`
  opacity: 1;
  #ticker {
    font-size: 2rem;
    margin-top: 90px;
    color: #fff;
    font-weight: 400;
    min-height: 56.667px;
  }
  .sub-heading {
    font-size: 26px;
    font-weight: 400;
    color: #fff;
  }
  &.features-banner {
    background-image: url(./../../static/images/feature-banner.jpg) !important;
  }
`;
