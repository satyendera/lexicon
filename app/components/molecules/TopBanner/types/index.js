// @flow

export type Props = {
  className: string,
  title: string,
  subTitle?: string,
  subHeading?: String,
  tickerPoints?: Array<string>,
};
