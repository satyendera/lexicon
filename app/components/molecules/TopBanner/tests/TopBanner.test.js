import { shallow } from 'enzyme';
// import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { TopBanner } from '../TopBanner';

describe('<TopBanner />', () => {
  let TopBannerComponent = '';
  beforeEach(() => {
    TopBannerComponent = shallow(<TopBanner title="Banner Title" subTitle="Banner subtitle" />);
  });

  test('should render correctly', () => {
    expect(TopBannerComponent).toMatchSnapshot();
  });

  test('should render correctly without subtitle', () => {
    TopBannerComponent.setProps({ subTitle: undefined });
    expect(TopBannerComponent).toMatchSnapshot();
  });
});
