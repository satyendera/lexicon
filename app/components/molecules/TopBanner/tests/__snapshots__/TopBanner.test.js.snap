// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopBanner /> should render correctly 1`] = `
<section
  id="banner"
>
  <div
    className="inner"
  >
    <h1>
      Banner Title
    </h1>
    <p>
      Banner subtitle
    </p>
  </div>
</section>
`;

exports[`<TopBanner /> should render correctly without subtitle 1`] = `
<section
  id="banner"
>
  <div
    className="inner"
  >
    <h1>
      Banner Title
    </h1>
  </div>
</section>
`;
